gateway/handler.go

```diff
@@ -358,6 +385,7 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	start := time.Now()
 	mrw := httputil.NewMetricResponseWriter(w)
+	setContentType(mrw, r)
```

Comment:
Won't this return `application/xml` for non-xml responses? (i.e. `GetObject`?)

Reply:
> Won't this return `application/xml` for non-xml responses? (i.e. `GetObject`?)

Ouch, good catch!  I'll need to figure out how to make go-swagger prefer application/xml :-(